cmd/deploy_test.go

```diff
@@ -1533,3 +1668,156 @@ func Test_ValidateBuilder(t *testing.T) {
 		t.Fatalf("did not get expected error validating an invalid builder name")
 	}
 }
+
+// TestReDeploy_OnRegistryChange tests that after deployed image with registry X,
+// subsequent deploy with registry Y triggers build
+func TestReDeploy_OnRegistryChange(t *testing.T) {
+	root := fromTempDirectory(t)
+
+	// Create a basic go Function
+	f := fn.Function{
+		Runtime: "go",
+		Root:    root,
+	}
+	_, err := fn.New().Init(f)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// create build cmd
+	cmdBuild := NewBuildCmd(NewTestClient(fn.WithBuilder(mock.NewBuilder())))
+	cmdBuild.SetArgs([]string{"--registry=" + TestRegistry})
+
+	// First: prebuild Function
+	if err := cmdBuild.Execute(); err != nil {
+		t.Fatal(err)
+	}
+
+	// change registry and deploy again
+	newRegistry := "example.com/fred"
+
+	cmd := NewDeployCmd(NewTestClient(
+		fn.WithDeployer(mock.NewDeployer()),
+	))
+
+	cmd.SetArgs([]string{"--registry=" + newRegistry})
+
+	// Second: Deploy with different registry and expect new build
+	if err := cmd.Execute(); err != nil {
+		t.Fatal(err)
+	}
+
+	// ASSERT
+	expectF, err := fn.NewFunction(root)
+	if err != nil {
+		t.Fatal("couldnt load function from path")
+	}
+
+	if !strings.Contains(expectF.Build.Image, newRegistry) {
+		t.Fatalf("expected built image '%s' to contain new registry '%s'\n", expectF.Build.Image, newRegistry)
+	}
+}
+
+// TestReDeploy_OnRegistryChangeWithBuildFalse should fail with function not
+// being built because the registry has changed
+func TestReDeploy_OnRegistryChangeWithBuildFalse(t *testing.T) {
```

Comment:
```suggestion
// TestDeploy_RegistryUpdateRequiresBuild should fail with function not
// being built because the registry has changed
func TestDeploy_RegistryUpdateRequiresBuild(t *testing.T) {
```
(I'm making these name change suggestions simply so that all of the deploy tests sort alphanumerically and can be run with `go test -run TestDeploy_`, even the ReDeploys)